pkg/operation/botanist/component/metricsserver/metrics_server.go

```diff
@@ -62,32 +64,37 @@ const (
 func New(
 	client client.Client,
 	namespace string,
-	secretsManager secretsmanager.Interface,
-	image string,
-	vpaEnabled bool,
-	kubeAPIServerHost *string,
+	values Values,
 ) component.DeployWaiter {
 	return &metricsServer{
-		client:            client,
-		namespace:         namespace,
-		secretsManager:    secretsManager,
-		image:             image,
-		vpaEnabled:        vpaEnabled,
-		kubeAPIServerHost: kubeAPIServerHost,
+		client:    client,
+		namespace: namespace,
+		values:    values,
 	}
 }
 
+// Values is a set of configuration values for the metrics server component.
+type Values struct {
+	// Image is the container image used for the metrics server.
+	Image string
+	// SecretsManager is the manager responsible for managing metrics server's secrets.
+	SecretsManager secretsmanager.Interface
```

Comment:
Do we have other example where the `SecretsManager` is part of the `Values` struct of a component? I would rather move the `SecretsManager` to be part of the `metricsServer` struct (as before).